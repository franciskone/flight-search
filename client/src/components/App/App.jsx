import React from 'react';

import STYLES from './App.scss';
import Header from '../Header';
import SearchResults from '../SearchResults/';

const c = className => STYLES[className] || 'UNKNOWN';

const App = () => (
  <div className={c('App')}>
    <Header />
    <main className={c('App__main')}>
      <SearchResults />
    </main>
  </div>
);

export default App;
