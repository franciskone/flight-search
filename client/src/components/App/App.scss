@import '~bpk-mixins';

// TODO Franciskone: understand if I still need this style
.App {
  &__main {}
}
