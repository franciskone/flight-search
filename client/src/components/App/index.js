import AppWithStore from './App';

export default AppWithStore;
