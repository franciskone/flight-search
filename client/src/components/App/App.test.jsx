import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import { App } from './App';

// TODO Franciskone: add https://github.com/dmitry-zaets/redux-mock-store
describe('App', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<App />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<App />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

