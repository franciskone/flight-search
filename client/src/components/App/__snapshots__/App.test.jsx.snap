// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <Header />
  <main
    className="App__main"
  >
    <SearchResults />
  </main>
</div>
`;
